### PR TITLE
feat: Validate long and short names for a single route

### PR DESCRIPTION
**Pull request by aababilov**
* RouteBothShortAndLongNameMissingNotice - Both short_name and long_name are missing for a route.
* RouteShortAndLongNameEqualNotice - Short and long name are equal for a route.
* RouteShortNameTooLongNotice - Short name of a route is too long (more than 6 characters).

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Validates short and long name for a single route.
+ * <p>
+ * Generated notices:
+ * * RouteBothShortAndLongNameMissingNotice
+ * * RouteShortAndLongNameEqualNotice
+ * * RouteShortNameTooLongNotice
+ */
+@GtfsValidator
+public class RouteNameValidator extends SingleEntityValidator<GtfsRoute> {
+    private static final int MAX_SHORT_NAME_LENGTH = 6;
```
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java:36]: Spec best practices say max of 12 characters:
https://gtfs.org/best-practices/#routestxt

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteShortNameTooLongNotice.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.notice;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Short name of a route is too long (more than 6 characters).
```
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteShortNameTooLongNotice.java:22]: Spec best practices say max of 12 characters:
https://gtfs.org/best-practices/#routestxt

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        final boolean hasLongName = entity.hasRouteLongName();
+        final boolean hasShortName = entity.hasRouteShortName();
+
+        if (!hasLongName && !hasShortName) {
+          noticeContainer.addNotice(new RouteBothShortAndLongNameMissingNotice(
+              entity.routeId(), entity.csvRowNumber()));
+        }
+
+        if (hasShortName && hasLongName &&
+            entity.routeShortName().equalsIgnoreCase(entity.routeLongName())) {
```
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java:49]: This rule could be expanded from "equals" to "contains" - the [spec best practices](https://gtfs.org/best-practices/#routestxt) say that:
>route_long_name should not contain the route_short_name.

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.notice.*;
```
Comment (barbeau) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java:22]: The wild card snuck in again ;)

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        assertThat(validateRoute(createRoute("SA", "Sa")))
+                .containsExactly(
+                        new RouteShortAndLongNameEqualNotice("r1", 1, "SA", "Sa"));
+
+        assertThat(validateRoute(createRoute("S", "Long"))).isEmpty();
+    }
+
+    @Test
+    public void routeShortNameTooLong() {
+        assertThat(validateRoute(createRoute("THISISMYSHORTNAME", null)))
```
Comment (barbeau) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java:71]: Should include assertion testing for false positive - when short name is less than max length.

**Review by barbeau — CHANGES_REQUESTED** [review]
Thanks @aababilov! A few comment in-line.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java (hunk 1 above)
Comment (aababilov) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java:36]: Indeed! And Google uses 6 characters internally. OK, let's use the public best practices.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteShortNameTooLongNotice.java (hunk 2 above)
Comment (aababilov) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/notice/RouteShortNameTooLongNotice.java:22]: Fixed.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java (hunk 3 above)
Comment (f8full) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java:49]: Warming though that can be tricky to find a good heuristic. Some long name contain the short one for good reasons in some network. I guess it's ok for a warning is we're doing something basic.

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java (hunk 5 above)
Comment (aababilov) [reply main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java:71]: Added.

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java (hunk 4 above)
Comment (aababilov) [reply main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java:22]: Oh yeah, I wrote this commit before configuring IDEA to use no wildcards. Fixed.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java (hunk 3 above)
Comment (aababilov) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java:49]: This best practice may be hard to check and follow: consider `route_short_name` = 'C' and `route_long_name` = 'Central'. What we have at Google is:

* check that long and short name are different;
* check that long name does not start from the short name followed by ' ', '-' or '('.

The second check is tricky and we can add it later.

Let's start from just comparing long and short names.

Comment (barbeau) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java:49]: Good points - ok, let's stick with equals for now.

**Review by barbeau — APPROVED** [review]

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java (hunk 3 above)
Comment (barbeau) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java:49]: Actually, we already have a six-month-old issue for this that I forgot about 😆  - https://github.com/MobilityData/gtfs-validator/issues/267

**Pull request merged**